core/conversion/converters/impl/expand.cpp

```diff
@@ -76,77 +146,165 @@ bool add_expand(ConversionCtx* ctx, const torch::jit::Node* n, nvinfer1::ITensor
   return true;
 }
 
+bool add_expand_dynamic(
+    ConversionCtx* ctx,
+    const torch::jit::Node* n,
+    nvinfer1::ITensor* in,
+    nvinfer1::ITensor* expandedDimsTensor) {
+  auto input_shape_tensor = ctx->net->addShape(*in)->getOutput(0);
+  auto input_rank = in->getDimensions().nbDims;
+  auto output_rank = expandedDimsTensor->getDimensions().d[0];
+  TRTORCH_CHECK(
+      input_rank <= output_rank,
+      "Number of dimensions of the desired expansion must be greater than or equal to the number of input dimensions");
+
+  // add a plugin to check expandedDimsTensor whether match input_shape_tensor
+  auto expandShape_layer =
+      create_plugin(ctx, n, input_shape_tensor, expandedDimsTensor, input_rank, output_rank, "expandShape");
+  auto _tensor = expandShape_layer->getOutput(0);
+
+  size_t max_rank = std::max(input_rank, output_rank);
+
+  // Dimensions are right alignment
+  auto new_input_shape_tensor = concat(max_rank, input_rank, ctx, in);
+  // LOG_DEBUG("Expand layer output tensor shape: " << new_output_shape_tensor->getDimensions());
+  auto new_output_shape_tensor = expandedDimsTensor;
+
+  // Add a reshape layer to expand dims
+  auto shuffle = ctx->net->addShuffle(*in);
+  shuffle->setInput(1, *new_input_shape_tensor);
+
+  // Start the slicing from beginning of tensor since this is an expand layer
+  std::vector<int64_t> start_vec(max_rank, 0);
+  nvinfer1::Dims starts_dim = util::toDims(c10::IntArrayRef(start_vec));
+
+  // compute sizes = max(x,y).
+  auto sizes =
+      ctx->net->addElementWise(*new_input_shape_tensor, *new_output_shape_tensor, nvinfer1::ElementWiseOperation::kMAX)
+          ->getOutput(0);
+  nvinfer1::Dims sizes_dim{-1, {}};
+  sizes_dim.nbDims = max_rank;
+
+  // Compute (x > 1 ? 1 : 0) for x in newDims, assuming positive x, using only TensorRT operations.
+  // min(1, sub(input_shape, 1))
+  int32_t* one_vector_tmp = new int32_t[1];
+  one_vector_tmp[0] = 1;
+  auto one_vector = vec2Tensor(one_vector_tmp, 1, ctx);
+  auto x_sub_one = ctx->net->addElementWise(*new_input_shape_tensor, *one_vector, nvinfer1::ElementWiseOperation::kSUB)
+                       ->getOutput(0);
+  auto strides = ctx->net->addElementWise(*one_vector, *x_sub_one, nvinfer1::ElementWiseOperation::kMIN)->getOutput(0);
+  nvinfer1::Dims strides_dim{-1, {}};
+  strides_dim.nbDims = max_rank;
+
+  // Slice layer does the expansion in TRT. Desired output size is specified by expandedDimsTensor

```

Comment:
We can modify the comment here. Desired output size is specified by sizes input at index 2.